tests/foreman/ui/test_contentviews.py

```diff
@@ -1233,3 +1240,92 @@ def test_delete_version(self):
             session.nav.go_to_content_views()
             self.content_views.delete_version(cv.name, version)
             self.content_views.validate_version_deleted(cv.name, version)
+
+    def test_delete_version_non_default(self):
+        """@Test: Delete a content-view version associated to non-default
+        environment
+
+        @Assert: Deletion fails
+
+        @Feature: ContentViewVersion
+
+        """
+        org = entities.Organization().create()
+        product_id = entities.Product(organization=org.id).create_json()['id']
+        repo = entities.Repository(product=product_id).create()
+        repo.sync()
+        cv = entities.ContentView(
+            name=gen_string('alpha'),
+            organization=org.id,
+            ).create()
+        cv.set_repository_ids([repo.id])
+        cv.publish()
+
+        cv_info = cv.read_json()['versions'][0]
+        version = 'Version {0}'.format(cv_info['version'])
+        cvv = entities.ContentViewVersion(
+            id=cv_info['id']
+        )
+        lc_env_id = entities.LifecycleEnvironment(
+            organization=org.id
+        ).create_json()['id']
+        cvv.promote(lc_env_id)
+
+        with Session(self.browser) as session:
+            session.nav.go_to_select_org(org.name)
+            session.nav.go_to_content_views()
+            self.content_views.delete_version(cv.name, version)
+            self.content_views.validate_version_deleted(cv.name, version)
+
+    def test_delete_version_with_ak(self):
+        """@Test: Delete a content-view version that had associated activation
+        key to it
+
+        @Assert: Deletion fails
+
+        @Feature: ContentViewVersion
+
+        """
+
+        org = entities.Organization().create()
+        cv = entities.ContentView(
+            name=gen_string('alpha'),
+            organization=org.id,
+            ).create()
+        cv.publish()
+
+        cv_info = cv.read_json()['versions'][0]
+        version = 'Version {0}'.format(cv_info['version'])
+        cvv = entities.ContentViewVersion(
+            id=cv_info['id']
+        )
+        lc_env = entities.LifecycleEnvironment(
+            organization=org.id
+        ).create_json()
+        cvv.promote(lc_env['id'])
+
+        ak = entities.ActivationKey(
+            name=gen_string('alphanumeric'),
+            environment=lc_env['id'],
+            organization=org.id,
+            content_view=cv.id,
+        ).create_json()
+
+        with Session(self.browser) as session:
+            session.nav.go_to_select_org(org.name)
+            session.nav.go_to_content_views()
+            self.content_views.validate_version_cannot_be_deleted(
+                cv.name,
+                version,
+            )
+            session.nav.go_to_dashboard()
```

Comment:
That call was made to prevent ```StaleElementException``` in our search functionality as our code do not differentiate search field locator for ActivationKeys or ContentView or any other possible entities in our application
.Definitely, it is a more workaround, but any other solution will be very time consuming and will initiate too many unnecesary changes to the code